ciphers/hill_cipher.py

```diff
@@ -42,6 +42,10 @@
 
 
 def gcd(a, b):
```

Comment:
Please add type hints to function parameters and return values.  This is where type hints are most valuable/helpful because in other locations CPython is often smart enough to determine the type from the context.
```suggestion
def gcd(a: int, b: int) -> int
```